src/main/java/fi/otavanopisto/kuntaapi/server/integrations/ptv/PtvServiceEntityUpdater.java

```diff
@@ -131,15 +138,29 @@ private void updatePtvService(ServiceId serviceId, Long orderIndex) {
         identifier = identifierController.updateIdentifier(identifier, null, orderIndex);
       }
       
-      Service service = response.getResponse();
-
+      Service ptvService = response.getResponse();
+      ServiceId kuntaApiServiceId = new ServiceId(KuntaApiConsts.IDENTIFIER_NAME, identifier.getKuntaApiId());
+      StatutoryDescription ptvStatutoryDescription = ptvService.getStatutoryDescriptionId() != null ? getStatutoryDescription(ptvService.getStatutoryDescriptionId()) : null;
+      
+      fi.metatavu.kuntaapi.server.rest.model.Service service = ptvTranslator.translateService(kuntaApiServiceId, ptvService, ptvStatutoryDescription);
+      ptvServiceCache.put(kuntaApiServiceId, service);
       modificationHashCache.put(identifier.getKuntaApiId(), createPojoHash(service));
-      index(identifier.getKuntaApiId(), service);
+      index(identifier.getKuntaApiId(), ptvService);
     } else {
       logger.warning(String.format("Service %s processing failed on [%d] %s", serviceId.getId(), response.getStatus(), response.getMessage()));
     }
   }
   
+  private StatutoryDescription getStatutoryDescription(String statutoryDescriptionId) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused private "getStatutoryDescription" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)